presto-main/src/main/java/io/prestosql/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -149,7 +153,79 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
             remainingSplits = splits;
         }
 
-        for (Split split : remainingSplits) {
+        Set<Split> remainingSplitsAfterBatchScheduling;
+        if (optimizedLocalScheduling && optimizedBatchScheduling) {
+            // only schedule split can be accessed remotely
+            remainingSplitsAfterBatchScheduling = remainingSplits.stream().filter(split -> !split.isRemotelyAccessible()).collect(Collectors.toSet());
+            Set<Split> remoteSplits = remainingSplits.stream().filter(split -> split.isRemotelyAccessible()).collect(Collectors.toSet());
+            long numOfRemoteSplits = remainingSplits.size();
```

Comment:
We avoid abbreviations in Presto, because we believe that not having them makes the code more readable for those new to Presto and for non-native english users.  In this case you can use `remoteSplitsCount` or simply inline this since the variable name doesn't add much value.